### PR TITLE
Add “Using with React” doc

### PR DESCRIPTION
**Pull request by dangodev**
## Reason for change

Adds a React doc with some gotchas / examples of integrating. Came up in a conversation with Leah and Kirsten today.

Also updates deps

Screen Shot 2019-10-03 at 16 31 07

## Checklist

- [x] **CHANGELOG**: The **Unreleased** section of CHANGELOG was updated
- [x] **Prop changes**: [docs][docs] were updated
- [x] **Prop changes**: E2E tests were updated, testing from the highest level possible
- [ ] **Platform testing**: If this change should be [tested against a platform][platform-testing], has it been?

[docs]: https://ui.sandbox.manifold.co
[platform-testing]: https://app.gitbook.com/@manifold/s/engineering/playbooks/testing-your-code-against-a-platform

### CODE REVIEW

File: docs/package.json [hunk 1]
```diff
@@ -14,47 +14,46 @@
...
+    "stylelint": "^11.0.0",
+    "stylelint-config-prettier": "^6.0.0",
     "stylelint-config-rational-order": "^0.1.2",
     "stylelint-config-standard": "^18.3.0",
     "stylelint-config-styled-components": "^0.1.1",
-    "stylelint-order": "^3.0.0",
-    "stylelint-processor-styled-components": "^1.6.0"
+    "stylelint-order": "^3.1.1",
+    "stylelint-processor-styled-components": "^1.8.0"
   }
```
Comment (dangodev) [inline docs/package.json]: Updates deps (dependabot doesn’t scan subdirectories)

File: docs/docs/advanced/react-events.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+title: Using with React
+path: /advanced/using-with-react
+---
+
+# Using with React
+
+Since UI are standard web components, that means React support is fully-built in since web
+components are now a fully-supported browser standard. However, if you’re new to web components,
+there can be some gotchas. Here are the tough parts we’ve found while implementing UI in a React
+environment.
```
Comment (davidleger95) [inline docs/docs/advanced/react-events.md]: Don't accept the suggestion in GH because the diff is wrong.

Just some more concise wording that you can take or leave.

```suggestion
Since Manifold UI components are standard Web Components, React is fully-supported. However, 
if you’re new to Web Components, there can be some gotchas. Here are some of the challenges 
we've experienced while implementing Manifold UI with React.
```

File: docs/docs/advanced/react-events.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+## Rule 1: Events must be handled with listeners
+
+Handling web component events in React is arguably the hardest part. In React, you may be used to
+handy `onClick={() => }` props for easy callbacks. Unfortunately that’s not yet possible with web
+components. Listeners are required. In the React projects we’ve worked in, we’ve found [React
+Hooks][hooks] (specifically `useState` and `useRef`) to be the simplest form of handling this.
```
Comment (davidleger95) [inline docs/docs/advanced/react-events.md]:
```suggestion
Hooks][hooks] (specifically `useState` and `useRef`) helpful for managing event listeners on Web Components.
```

File: docs/docs/advanced/react-events.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+### Example
+
+Say you’re integrating the [Plan Selector][plan-selector] component in React, and you want to be
+notified every time a user changes their plan to pass to the [provision-button]. This is what that
```
Comment (davidleger95) [inline docs/docs/advanced/react-events.md]:
```suggestion
notified when the plan changes in order to to pass the selected plan to the [provision-button]. This is what that
```

File: docs/docs/advanced/react-events.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+### How do I find what events are possible?
+
+Most Manifold UI components emit several events. To see what each component emits, please reference
+the **Events** section of that component’s documentation.
+
+## Rule 2: Every prop is a string
+
+Another reason consuming web component events can be difficult within React is because whereas React
+has no problem passing any data type (`object`, `array`, `number`, etc.) as a prop, **web components
+can only accept strings** as they’re HTML:
```
Comment (davidleger95) [inline docs/docs/advanced/react-events.md]: Don't accept suggestion in GH. Diff is wrong

```suggestion
Another reason consuming web component events can be difficult within React is because **web components can only accept strings as props** because they’re just regular HTML elements:
```

**Comment by davidleger95** [comment]
You may notice I used `Web Components` instead of `web components` in some places. There is a question in the community around whether to capitalize this term. I'm in favor of capitalization because it makes it clearer that you're talking about the web standard rather than just components that happen to be on the web. Often when I talk about Web Components to people they assume React/Vue/Angular/etc. components. This is a subtle differentiation, but I think it helps emphasize that there is a difference.

**Comment by dangodev** [comment]
@davidleger95 Yeah I’m fine with capitalizing Web Components. I don’t feel strongly, but the capitalization (at least in 2019) does help signal we’re referring to a specific thing. Perhaps the usage will change over time but capitalizing seems like the right call for now.
